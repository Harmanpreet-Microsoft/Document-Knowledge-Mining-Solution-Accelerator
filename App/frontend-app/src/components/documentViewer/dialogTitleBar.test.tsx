import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { DialogTitleBar } from "./dialogTitleBar";

// Mock the dependencies
jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key,
  }),
}));

jest.mock("../chat/chatRoom", () => ({
  ChatRoom: () => <div data-testid="chat-room">Chat Room Component</div>,
}));

describe("DialogTitleBar Component", () => {
  const defaultProps = {
    handleDialogClose: jest.fn(),
    metadata: { title: "Document Title", source_last_modified: "2023-12-01" },
    selectedPage: null,
    selectedTab: "Document",
    onTabSelect: jest.fn(),
    pageMetadata: null,
    selectedPageMetadata: null,
    handleReturnToDocumentTab: jest.fn(),
    downloadFile: jest.fn(),
    urlWithSasToken: "https://example.com",
    styles: {},
    props: {},
    clearChatFlag: true,
    setClearChatFlag: jest.fn(),
  };

  it("renders without crashing", () => {
    render(<DialogTitleBar {...defaultProps} />);
    expect(screen.getByText("Document Title")).toBeInTheDocument();
    expect(screen.getByText("2023-12-01")).toBeInTheDocument();
  });

  it("calls handleDialogClose when the close button is clicked", () => {
    render(<DialogTitleBar {...defaultProps} />);
    const closeButton = screen.getByLabelText("close");
    fireEvent.click(closeButton);
    expect(defaultProps.handleDialogClose).toHaveBeenCalled();
  });

  it("calls onTabSelect when a tab is selected", () => {
    render(<DialogTitleBar {...defaultProps} />);
    const documentTab = screen.getByText("components.dialog-title-bar.document");
    fireEvent.click(documentTab);
    expect(defaultProps.onTabSelect).toHaveBeenCalled();
  });

  it("renders correctly when metadata is incomplete", () => {
    render(<DialogTitleBar {...defaultProps} metadata={{}} />);
    expect(screen.queryByText("Document Title")).not.toBeInTheDocument();
    expect(screen.queryByText("2023-12-01")).not.toBeInTheDocument();
  });

  it("does not render the 'Pages' tab when pageMetadata is empty", () => {
    render(<DialogTitleBar {...defaultProps} pageMetadata={[]} />);
    expect(screen.queryByText("components.dialog-title-bar.pages")).not.toBeInTheDocument();
  });

  it("renders the correct tab content for 'AI Knowledge'", () => {
    render(<DialogTitleBar {...defaultProps} selectedTab="AI Knowledge" />);
    expect(screen.getByText("components.dialog-title-bar.ai-knowledge")).toBeInTheDocument();
  });

  it("renders close button with aria-label", () => {
    render(<DialogTitleBar {...defaultProps} />);
    const closeButton = screen.getByLabelText("close");
    expect(closeButton).toBeInTheDocument();
  });

  it("disables the download button when urlWithSasToken is undefined", () => {
    render(<DialogTitleBar {...defaultProps} urlWithSasToken={undefined} />);
    const downloadButton = screen.getByText("components.dialog-title-bar.download");
    expect(downloadButton).toBeDisabled();
  });

  it("enables the download button when urlWithSasToken is provided", () => {
    render(<DialogTitleBar {...defaultProps} />);
    const downloadButton = screen.getByText("components.dialog-title-bar.download");
    expect(downloadButton).toBeEnabled();
  });

  it("calls downloadFile when the download button is clicked", () => {
    render(<DialogTitleBar {...defaultProps} />);
    const downloadButton = screen.getByText("components.dialog-title-bar.download");
    fireEvent.click(downloadButton);
    expect(defaultProps.downloadFile).toHaveBeenCalled();
  });

  it("renders the ChatRoom component when the 'Chat Room' tab is active", () => {
    render(<DialogTitleBar {...defaultProps} selectedTab="Chat Room" />);
    expect(screen.getByTestId("chat-room")).toBeInTheDocument();
  });

  it("calls handleReturnToDocumentTab when the return button is clicked", () => {
    render(
      <DialogTitleBar
        {...defaultProps}
        selectedTab="Page Number"
        selectedPage={1}
        selectedPageMetadata={{ tables: [1] }}
      />
    );
    const returnButton = screen.getByText("components.dialog-title-bar.return-to-document");
    fireEvent.click(returnButton);
    expect(defaultProps.handleReturnToDocumentTab).toHaveBeenCalled();
  });

  it("matches snapshot for the default state", () => {
    const { asFragment } = render(<DialogTitleBar {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});