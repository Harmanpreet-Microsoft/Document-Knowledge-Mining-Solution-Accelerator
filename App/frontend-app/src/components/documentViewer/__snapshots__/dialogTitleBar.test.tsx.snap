// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogTitleBar Component matches snapshot for the default state 1`] = `
<DocumentFragment>
  <h2
    class="fui-DialogTitle rztv7rx"
    style="width: 100%;"
  >
    <div>
      Document Title
    </div>
    <div>
      <span
        class="fui-Text ___1iu0j08_1xl236j fk6fouc fy9rknc fwrc4pm fl43uef fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
        style="color: rgb(77, 77, 77);"
      >
        2023-12-01
      </span>
    </div>
    <div
      class="-ml-3 mt-5 flex justify-between"
    >
      <div>
        <div
          aria-orientation="horizontal"
          class="fui-TabList ___8n9zdb0_1l1dwte f22iagw f1063pyq fi64zpg flvyvdh f10pi13n f1q9h2pe"
          data-tabster="{"mover":{"cyclic":true,"direction":2,"memorizeCurrent":true}}"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="fui-Tab ___yw01pi0_1ki54am f122n59 fwhevhj f61n433 fv1dfc8 f1q8l70w f1aa9q02 f16jpd5f f1jar5jt fyu767a fre7gi1 f1358rze fqdk4by f1rvrf73 f1k6fduh f13qh94s fi64zpg f1u07yai frn2hmy f1olsevy fk6fouc f1i3iumi f1s6fcnf f10pi13n f1mtd64y f1y7q3j9 f1cxpek8 f4d9j23 f1rjii52 f5yzyt f11gcy0p fx3omr f1ng84yb f8hki3x f1d2448m f1bjia2o ffh67wi f1p7hgxw f1way5bb f9znhxp fqa318h f1vjpng2 f1c21dwh fjxutwb fophhak f1cadz5z f1ck17l f42ak0g ffplhdr ffth601 fhklyu5 fhw179n f10y1uxy f6aiuy0 f1kfpfnu f1dx5wco f1kmhr4c fl1ydde f1y7maxz f105swax fscdmel f1tkcw1w f1u11x8o f16cxu0 f1nwgacf f15ovonk fvje46l fchca7p f1s2r9ax f1o0nnkk fxb7rol f1ksivud fprarqb f14vs0nd f1gtfqs9 f18zvfd9 f13zj6fq f1mdlcz9 ffmd2fr f1eytvvh fuaa9s fy7ktjt f16tp0gf fj0yp7j fo72kxq f1vx7lu8 fna7m5n f1oxpfwv fhwpy7i f6zz20j f1ai4sc1 f9qxlq5 f1aql376 fug4aj8 f1q7ujh"
            role="tab"
            style="--fui-Tab__indicator--offset: 0px; --fui-Tab__indicator--scale: 1;"
            type="button"
            value="Document"
          >
            <span
              class="fui-Tab__icon ___qr1fdx0_el26xc0 fwpfdsa f16hsg94 f122n59 ftuwxu6 f4d9j23 f1p9o1ba f1sil6mw fxoiby5 f15q0o9g fe5j1ua fjamq6b f64fuq3"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 10.5a.5.5 0 1 1 1 0 .5.5 0 0 1-1 0Zm.5 1.5a.5.5 0 1 0 0 1 .5.5 0 0 0 0-1ZM6 14.5a.5.5 0 1 1 1 0 .5.5 0 0 1-1 0ZM8.5 10a.5.5 0 0 0 0 1h5a.5.5 0 0 0 0-1h-5ZM8 12.5c0-.28.22-.5.5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5Zm.5 1.5a.5.5 0 0 0 0 1h5a.5.5 0 0 0 0-1h-5ZM6 2a2 2 0 0 0-2 2v12c0 1.1.9 2 2 2h8a2 2 0 0 0 2-2V7.41c0-.4-.16-.78-.44-1.06l-3.91-3.91A1.5 1.5 0 0 0 10.59 2H6ZM5 4a1 1 0 0 1 1-1h4v3.5c0 .83.67 1.5 1.5 1.5H15v8a1 1 0 0 1-1 1H6a1 1 0 0 1-1-1V4Zm9.8 3h-3.3a.5.5 0 0 1-.5-.5V3.2L14.8 7Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span
              class="fui-Tab__content ___6i206w0_w0jjr10 fk6fouc fkhj508 fl43uef f1i3iumi f1p9o1ba f1sil6mw fztplxc ffczdla f9g1xly fgiv446 fd46tj4 f16hsg94"
            >
              components.dialog-title-bar.document
            </span>
          </button>
          <button
            aria-selected="false"
            class="fui-Tab ___1ok5per_112k432 f122n59 fwhevhj f61n433 fv1dfc8 f1q8l70w f1aa9q02 f16jpd5f f1jar5jt fyu767a fre7gi1 f1358rze fqdk4by f1rvrf73 f1k6fduh f13qh94s fi64zpg f1u07yai frn2hmy f1olsevy fk6fouc f1i3iumi f1s6fcnf f10pi13n f1p9o1ba f1sil6mw f1cxpek8 f4d9j23 f1rjii52 f5yzyt f11gcy0p fx3omr f1ng84yb f8hki3x f1d2448m f1bjia2o ffh67wi f1p7hgxw f1way5bb f9znhxp fqa318h f1vjpng2 f1c21dwh fjxutwb fophhak fmmjozx f1mfqf41 f10aiid4 fpkze5g f149wc3x fjioou7 fhw179n f10y1uxy f6aiuy0 f1kfpfnu f1dx5wco f1kmhr4c fl1ydde f1y7maxz f105swax fscdmel f1tkcw1w f1u11x8o f16cxu0 f1nwgacf f15ovonk fvje46l fchca7p f1s2r9ax f1o0nnkk fxb7rol"
            role="tab"
            style="--fui-Tab__indicator--offset: 0px; --fui-Tab__indicator--scale: 1;"
            type="button"
            value="AI Knowledge"
          >
            <span
              class="fui-Tab__icon ___v0u0aj0_1ro3490 fwpfdsa f16hsg94 f122n59 ftuwxu6 f4d9j23 f1p9o1ba f1sil6mw f16u1re f8bsbmo fe5j1ua fjamq6b f64fuq3"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.4 12.8a1.04 1.04 0 0 0 1.59-.51l.45-1.37a2.34 2.34 0 0 1 1.47-1.48l1.4-.45A1.04 1.04 0 0 0 12.25 7l-1.37-.45A2.34 2.34 0 0 1 9.4 5.08L8.95 3.7a1.03 1.03 0 0 0-.82-.68 1.04 1.04 0 0 0-1.15.7l-.46 1.4a2.34 2.34 0 0 1-1.44 1.45L3.7 7a1.04 1.04 0 0 0 .02 1.97l1.37.45a2.33 2.33 0 0 1 1.48 1.48l.46 1.4c.07.2.2.37.38.5Zm6.14 4.05a.8.8 0 0 0 1.22-.4l.25-.76a1.09 1.09 0 0 1 .68-.68l.77-.25a.8.8 0 0 0-.02-1.52l-.77-.25a1.08 1.08 0 0 1-.68-.68l-.25-.77a.8.8 0 0 0-1.52.01l-.24.76a1.1 1.1 0 0 1-.67.68l-.77.25a.8.8 0 0 0 0 1.52l.77.25a1.09 1.09 0 0 1 .68.68l.25.77c.06.16.16.3.3.4Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span
              class="fui-Tab__content ___1xjz13j_1ux7tna fk6fouc fkhj508 figsok6 f1i3iumi f1p9o1ba f1sil6mw fztplxc ffczdla f9g1xly fgiv446 fd46tj4 f16hsg94"
            >
              components.dialog-title-bar.ai-knowledge
            </span>
            <span
              class="fui-Tab__content--reserved-space ___1raik6b_obabww0 fk6fouc fkhj508 fl43uef f1i3iumi f1p9o1ba f1sil6mw fztplxc ffczdla f9g1xly fgiv446 fd46tj4 f16hsg94 fd7fpy0"
            >
              components.dialog-title-bar.ai-knowledge
            </span>
          </button>
          <button
            aria-selected="false"
            class="fui-Tab ___1ok5per_112k432 f122n59 fwhevhj f61n433 fv1dfc8 f1q8l70w f1aa9q02 f16jpd5f f1jar5jt fyu767a fre7gi1 f1358rze fqdk4by f1rvrf73 f1k6fduh f13qh94s fi64zpg f1u07yai frn2hmy f1olsevy fk6fouc f1i3iumi f1s6fcnf f10pi13n f1p9o1ba f1sil6mw f1cxpek8 f4d9j23 f1rjii52 f5yzyt f11gcy0p fx3omr f1ng84yb f8hki3x f1d2448m f1bjia2o ffh67wi f1p7hgxw f1way5bb f9znhxp fqa318h f1vjpng2 f1c21dwh fjxutwb fophhak fmmjozx f1mfqf41 f10aiid4 fpkze5g f149wc3x fjioou7 fhw179n f10y1uxy f6aiuy0 f1kfpfnu f1dx5wco f1kmhr4c fl1ydde f1y7maxz f105swax fscdmel f1tkcw1w f1u11x8o f16cxu0 f1nwgacf f15ovonk fvje46l fchca7p f1s2r9ax f1o0nnkk fxb7rol"
            role="tab"
            style="--fui-Tab__indicator--offset: 0px; --fui-Tab__indicator--scale: 1;"
            type="button"
            value="Chat Room"
          >
            <span
              class="fui-Tab__icon ___v0u0aj0_1ro3490 fwpfdsa f16hsg94 f122n59 ftuwxu6 f4d9j23 f1p9o1ba f1sil6mw f16u1re f8bsbmo fe5j1ua fjamq6b f64fuq3"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2a10 10 0 1 1-4.59 18.89L3.6 21.96a1.25 1.25 0 0 1-1.54-1.54l1.06-3.83A10 10 0 0 1 12 2Zm0 1.5a8.5 8.5 0 0 0-7.43 12.64l.15.27-1.1 3.98 3.98-1.11.27.15A8.5 8.5 0 1 0 12 3.5ZM8.75 13h4.5a.75.75 0 0 1 .1 1.5h-4.6a.75.75 0 0 1-.1-1.5h4.6-4.5Zm0-3.5h6.5a.75.75 0 0 1 .1 1.5h-6.6a.75.75 0 0 1-.1-1.5h6.6-6.5Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span
              class="fui-Tab__content ___1xjz13j_1ux7tna fk6fouc fkhj508 figsok6 f1i3iumi f1p9o1ba f1sil6mw fztplxc ffczdla f9g1xly fgiv446 fd46tj4 f16hsg94"
            >
              Chat
            </span>
            <span
              class="fui-Tab__content--reserved-space ___1raik6b_obabww0 fk6fouc fkhj508 fl43uef f1i3iumi f1p9o1ba f1sil6mw fztplxc ffczdla f9g1xly fgiv446 fd46tj4 f16hsg94 fd7fpy0"
            >
              Chat
            </span>
          </button>
        </div>
      </div>
      <div
        class="flex space-x-2"
      >
        <button
          class="fui-Button r1alrhcs h-8 ___1akj6hk_ih97uj0 ffp7eso f1p3nwhy f11589ue f1q5o8ev f1pdflbu f1phragk f15wkkf3 f1s2uweq fr80ssc f1ukrpxl fecsdlb f1rq72xc fnp9lpt f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1d6v5y2 f1rirnrt f1uu00uk fkvaka8 f1ux7til f9a0qzu f1lkg8j3 fkc42ay fq7113v ff1wgvm fiob0tu f1j6scgf f1x4h75k f4xjyn1 fbgcvur f1ks1yx8 f1o6qegi fcnxywj fmxjhhp f9ddjv3 f17t0x8g f194v5ow f1qgg65p fk7jm04 fhgccpy f32wu9k fu5nqqq f13prjl2 f1czftr5 f1nl83rv f12k37oa fr96u23"
          type="button"
        >
          components.dialog-title-bar.download
        </button>
      </div>
    </div>
    <div
      class="pb-4"
    >
      <div
        aria-orientation="horizontal"
        class="fui-Divider ___14drqcq_2rxt2r0 f122n59 f1ewtqcl f22iagw f1063pyq fqerorx f10pi13n fk6fouc fy9rknc figsok6 fwrc4pm f17mccla fyl8oag f16vkdww fhsnbul f1gw3sf2 f1ly5f7u f1s3tz6t f1wl9k8s f13zj6fq fkfq4zb f1vccso1 f1geml7w f1r7kh1m fjml6kk f16j7guv fx01ahm fl8d8yv fj1a37q fly5x3f f163fonl f51yk4v f13rof3u f8rth92 f6czdpx f1iyka9k fzynn9s fekrn8e f1kyqvp9 fesgyo"
        role="separator"
      />
    </div>
  </h2>
  <div
    class="fui-DialogTitle__action r13kcrze"
  >
    <button
      aria-label="close"
      class="fui-Button r1alrhcs ___1p24v7a_y4hy550 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl f1sbtcvk fwiuce9 fdghr9 f15vdbe4 fwbmr0d f44c6la"
      data-tabster="{"restorer":{"type":1}}"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="fui-Button__icon rywnvv2"
      >
        <svg
          aria-hidden="true"
          class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m4.4 4.55.07-.08a.75.75 0 0 1 .98-.07l.08.07L12 10.94l6.47-6.47a.75.75 0 1 1 1.06 1.06L13.06 12l6.47 6.47c.27.27.3.68.07.98l-.07.08a.75.75 0 0 1-.98.07l-.08-.07L12 13.06l-6.47 6.47a.75.75 0 0 1-1.06-1.06L10.94 12 4.47 5.53a.75.75 0 0 1-.07-.98l.07-.08-.07.08Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
